core/meta/inc/TIsAProxy.h

```diff
@@ -28,14 +28,22 @@ class TIsAProxy  : public TVirtualIsAProxy {
 private:
    template <typename T> using Atomic_t = std::atomic<T>;
 
+   // On testing with the data from the 250202_181_RECO.root and doing "just" serializing
+   // the value 8 was the sweet spot of performance.  With more slots, too much time is
+   // spent scanning the array of "last" seen and with less slots then the
+   // serialization induced by and/or the cost of executed `++fSubTypesReaders is slow
+   // down (noticeably) the streaming of branches with polymorphic containers.
+   static constexpr UInt_t fgMaxLastSlot = 8;
+
    const std::type_info     *fType;        //Actual typeid of the proxy
-   Atomic_t<TClass*>         fClass;       //Actual TClass
-   Atomic_t<void*>           fLast;        //points into fSubTypes map for last used values
-   Char_t                    fSubTypes[72];//map of known sub-types
-   mutable Atomic_t<UInt_t>  fSubTypesReaders; //number of readers of fSubTypes
+   TClass                   *fClass;       //Actual TClass
+   Atomic_t<void*>           fLasts[fgMaxLastSlot];   // points into fSubTypes map for last used values
```

Comment:
Do you have an alternative?

Reply:
To avoid false sharing, you can try storing each pointer far enough apart that they'd fall into different cache lines. This can be either done with padding, or by combining more fields into each slot, like making the slot store a full pair, not just the pointer to it. I don't know if this is accessed frequently enough to be a big issue, though, just noticing that this is very similar to the false sharing that happens in `RDataFrame` filters, which are indeed a big performance problem there.